internal/services/oracle/oracle_network_anchor_resource.go

```diff
@@ -0,0 +1,402 @@
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
+		},
+
+		"oci_backup_cidr_block": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsCIDR,
+		},
+
+		"oracle_dns_forwarding_endpoint_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"oracle_dns_listening_endpoint_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"oracle_to_azure_dns_zone_sync_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"dns_forwarding_rule": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			DiffSuppressFunc: func(k, old, new string, d *pluginsdk.ResourceData) bool {
+				// Always suppress diff if we’re reading existing resource, dns_forwarding_rule is not returning
+				if d.Id() != "" {
+					return true
+				}
+				return false
+			},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"domain_names": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"forwarding_ip_address": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.IsIPAddress,
+					},
+				},
+			},
+		},
+
+		"dns_listening_endpoint_allowed_cidr": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.IsCommaSeparatedCIDRs,
+		},
+
+		"tags":  commonschema.Tags(),
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
+	}
+}
+
+func (NetworkAnchorResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"dns_forwarding_rule_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"dns_forwarding_endpoint_ip_address": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"dns_forwarding_endpoint_nsg_rule_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"dns_listening_endpoint_ip_address": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"dns_listening_endpoint_nsg_rule_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (NetworkAnchorResource) ModelObject() interface{} {
+	return &NetworkAnchorResource{}
+}
+
+func (NetworkAnchorResource) ResourceType() string {
+	return "azurerm_oracle_network_anchor"
+}
+
+func (r NetworkAnchorResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.NetworkAnchors
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model NetworkAnchorResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
```

Comment:
This should follow the standard return value for decoding errors:
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```